Automattic/wp-calypso
# Site Transfers: Improve username input box design

Closes https://github.com/Automattic/dotcom-forge/issues/2645
## Proposed Changes

This PR improves the design of the username input box on [https://wpcalypso.wordpress.com/settings/start-site-transfer/{SimpleSite}](https://wpcalypso.wordpress.com/settings/start-site-transfer/%7BSimpleSite%7D) to make it consistent [with the Figma design.](1U910ArebC7gbLF1RtxUxE-fi-291_11406)

**Before**:

<img width="905" alt="Screenshot 2023-06-06 at 9 55 09 AM" src="https://github.com/Automattic/wp-calypso/assets/25575134/88e3038b-9e69-483a-bc3d-f5691593dd68">

<img width="895" alt="Screenshot 2023-06-06 at 9 58 07 AM" src="https://github.com/Automattic/wp-calypso/assets/25575134/cbdb67c5-eadf-477d-8ac5-e5ae8f924241">

**After**:

<img width="926" alt="Screenshot 2023-06-06 at 9 57 13 AM" src="https://github.com/Automattic/wp-calypso/assets/25575134/a879969e-d201-4b36-93a3-f3407da2be9a">

<img width="861" alt="Screenshot 2023-06-06 at 9 57 40 AM" src="https://github.com/Automattic/wp-calypso/assets/25575134/33ebd3ab-6461-4fae-b5f5-80f779f65815">



## Testing Instructions

1. Make sure you have a simple site
2. Navigate to [https://wpcalypso.wordpress.com/settings/start-site-transfer/{SimpleSite}](https://wpcalypso.wordpress.com/settings/start-site-transfer/%7BSimpleSite%7D)
3. Observe that the input box design is consistent with the screenshot (the line-height is 26px, the border colors are grey)
4. Type in the input field `admin` and click on `Search user and continue`
5. Observe that the box border turns red

## Pre-merge Checklist
